src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/PropertiesShouldNotBeWriteOnlyTests.cs

```diff
@@ -9,12 +10,680 @@ public class PropertiesShouldNotBeWriteOnlyTests : DiagnosticAnalyzerTestBase
+{
+    public class ClassWithReadableProperty
+    {
+        protected string field;
+        protected string AccessibleProperty2
+        {
+        get { return field; }
+        private set { field = value; }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_internal_Read_private_Write()
+        {
+            var code = @"
+using System;
+namespace GoodPropertiesShouldNotBeWriteOnlyTests
+{
+    public class GoodClassWithReadWriteProperty
+    {
+        protected string field;
+        internal string AccessibleProperty3
+        {
+            get { return field; }
+            private set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_protected_internal_Read_internal_Write()
+        {
+            var code = @"
+using System;
+namespace GoodPropertiesShouldNotBeWriteOnlyTests
+{
+    public class GoodClassWithReadWriteProperty
+    {
+        protected string field;
+        protected internal string AccessibleProperty4
+        {
+            get { return field; }
+            internal set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_public_Read_internal_Write()
+        {
+            var code = @"
+using System;
+namespace GoodPropertiesShouldNotBeWriteOnlyTests
+{
+    public class GoodClassWithReadWriteProperty
+    {
+        protected string field;
+        public string AccessibleProperty5
+        {
+            get { return field; }
+            internal set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_public_Read_protected_Write()
+        {
+            var code = @"
+using System;
+namespace DesignLibrary
+{
+    public class GoodClassWithReadWriteProperty
+    {
+        protected string field;
+        public string AccessibleProperty6
+        {
+            get { return field; }
+            protected set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_public_override_Write()
+        {
+            var code = @"
+using System;
+namespace GoodPropertiesShouldNotBeWriteOnlyTests
+{
+    protected string field;
+    public class DerivedClassWithReadableProperty : ClassWithReadableProperty
+    {
+        public override string ReadableProperty1
+        {
+            set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+// Cannot get below test to work, not sure where the issue is.
+//        [Fact]
+//        public void CSharp_CA1044Interface_Write()
+//        {
+//            var code = @"
+//using System;
+//namespace GoodPropertiesShouldNotBeWriteOnlyTests
+//{
+//    public interface IInterface
```

Comment:
The snippet of code you have here is expected to raise a violation because it has a public property that is writeonly. The test is verifying that there are no diagnostics (by leaving out the expected values). This should instead be VerifyCSharp(code, GetCSharpResultAt(...)). Also as I mention in another comment, this test should really be merged with the next one.